accepted/0000-link-packages-to-source-and-build.md

```diff
@@ -0,0 +1,514 @@
+- Should be maintained with >99.9% uptime so that developers are not blocked from publishing new packages.
+- Should allow future extensions to support centrally managed signing authorities such as certificates managed by an enterprise and inner source within air-gapped enterprise environments.
+- Buy-in from the broader open source community.
+
+## Non-goals
+- Support for private/inner-source packages to begin with.
+  - Our primary goal is to link packages hosted on the npm registry to the source and build it originated from, for this to be broadly useful, the source code and build need to be publicly available. We're committed to eventually supporting private use cases.
+- Support for self-hosted CI/CD systems.
+- Mitigate every possible supply chain attack against the npm registry. For example:
+  - Registry/mirror/proxy MITM attacks
+  - Denial of service attacks against the registry or client from a compromised mirror/proxy
+    - Rollback/fast-forward attacks (e.g. providing the wrong version for update)
+    - Freeze/endless data attacks (e.g. preventing updates)
+    - Extraneous dependencies attacks (e.g. tricking the client to install undeclared dependencies)
+  - Malicious maintainer (e.g. [event-stream](https://blog.npmjs.org/post/180565383195/details-about-the-event-stream-incident))
+  - Compromised login credentials (where 2FA can help)
+  - Typosquatting attacks (e.g. `lodahs`), [“dependency confusion”](https://medium.com/@alex.birsan/dependency-confusion-4a5d60fec610)
+  - Accidental vulnerabilities
+
+## Sigstore as signing solution
+The [Sigstore](https://www.sigstore.dev/) project has been selected as the solution to signing npm packages (see detailed explanation below). It is currently the only working solution that supports our key requirements:
+- Links packages hosted on the npm registry to the source and build they originated from (provenance information).
+- Supports signing identities that don't expose any personally identifiable information (PII) about maintainers, e.g. emails.
+- Avoids developer-managed keys (as there's no good way to offer trust to the community given the challenges of distributing public keys).
+
+Note that nothing in this current proposal precludes us from offering additional signing methods in the future as they become available and mature.
+
+## Risks of adopting Sigstore
+Sigstore infrastructure and tooling are currently not considered production ready and are run on a best-effort basis. The Sigstore project is working towards a [General Availability](https://blog.sigstore.dev/an-update-on-general-availability-5c5563d4e400) release later this year but we don't yet know what these guarantees will look like yet.
+
+As such, there are several risks of adopting Sigstore for npm that are worth calling out:
+- No buy-in from the broader open source npm community.
```

Comment:
tbh this seems like a potential dealbreaker

Reply:
I don't disagree, but IMO this isn't because there wouldn't be but more because there's been near zero exposure.